.circleci/main.yml

```diff
@@ -69,57 +69,57 @@ executors:
-              -shard=$CIRCLE_NODE_INDEX \
-              -total-shards=$CIRCLE_NODE_TOTAL \
-              -threads=2 \
-              -gotestsum-args="--format=testdox" \
-              -gotest-args="-parallel=6 -timeout=20m -cover -test.gocoverdir=\"$TEST_RESULTS\" -coverprofile=${TEST_RESULTS}/coverage.txt -covermode=atomic -coverpkg=./..."
-      - run:
-          name: Format code coverage
-          when: always
-          command: etc/testing/circle/format-coverage.sh
-      - store_artifacts: # upload test summary for display in Artifacts
-          path: "/tmp/test-results/"
-          destination: raw-test-output
-      - store_test_results: # upload test results for display in Test Summary
-          path: "/tmp/test-results/circle"
-      - codecov/upload:
-          file: /tmp/test-results/coverage.txt
-      - collect-test-results
-      - save_cache:
-          key: pach-go-unittest-mod-cache-v1-{{arch}}-{{ checksum "go.sum" }}
-          paths:
-            - /home/circleci/go/pkg/mod
+  # test-go:
+  #   machine:
+  #     image: << pipeline.parameters.machine_image >>
+  #   resource_class: large
+  #   parallelism: 5
+  #   environment:
+  #     TEST_RESULTS: /tmp/test-results
+  #   steps:
+  #     - checkout
+  #     - go/install:
+  #         version: << pipeline.parameters.go-version >>
+  #     - run: mkdir ${TEST_RESULTS}
+  #     - restore_cache:
+  #         keys:
+  #           - pach-go-unittest-mod-cache-v1-{{arch}}-{{ checksum "go.sum" }}
+  #     - run: go mod download
+  #     - run:
+  #         name: Collect tests to run
+  #         command: GOMAXPROCS=2 PACHYDERM_LOG_LEVEL="debug" go run ./src/testing/cmds/harness/test-collector/main.go -file=tests_to_run_unit.csv -procs=16
+  #         background: true
+  #     - run: go install gotest.tools/gotestsum@latest
+  #     - run: CGO_ENABLED=0 GOMAXPROCS=2 go install ./src/server/cmd/pachctl
+  #     - run: go install ./src/testing/match
+  #     - run: etc/testing/circle/install.sh
+  #     - run:
+  #         name: Run Go Unit Tests
+  #         no_output_timeout: 20m
+  #         command: |-
+  #           GOMAXPROCS=7 GOMEMLIMIT=8000MiB go run src/testing/cmds/harness/test-runner/main.go \
+  #             -file=tests_to_run_unit.csv \
+  #             -shard=$CIRCLE_NODE_INDEX \
+  #             -total-shards=$CIRCLE_NODE_TOTAL \
+  #             -threads=2 \
+  #             -gotestsum-args="--format=testdox" \
+  #             -gotest-args="-parallel=6 -timeout=20m -cover -test.gocoverdir=\"$TEST_RESULTS\" -coverprofile=${TEST_RESULTS}/coverage.txt -covermode=atomic -coverpkg=./..."
+  #     - run:
+  #         name: Format code coverage
+  #         when: always
+  #         command: etc/testing/circle/format-coverage.sh
+  #     - store_artifacts: # upload test summary for display in Artifacts
+  #         path: "/tmp/test-results/"
+  #         destination: raw-test-output
+  #     - store_test_results: # upload test results for display in Test Summary
```

Comment:
WRT the JSON output. it is unfortunate that we don't have it. I'd hope that we could just change the processing from `json.Unmarshal` to `xml.Unmarshal` or equivalent. However, my understanding is that JUnit reports are very different, and formatting as JUnit has some haky stuff going on under the hood. If you want to defer handling of the xml to another ticket I would not be upset. I think there may be some data-massaging required to get it working.